app/code/Magento/Catalog/Block/Product/ImageFactory.php

```diff
@@ -131,10 +131,10 @@ private function getLabel(Product $product, string $imageType): string
      *
      * @param Product $product
      * @param string $imageId
-     * @param array|null $attributes
+     * @param array $attributes
      * @return ImageBlock
      */
-    public function create(Product $product, string $imageId, array $attributes = null): ImageBlock
+    public function create(Product $product, string $imageId, array $attributes = []): ImageBlock
```

Comment:
Unfortunately, we are not allowed to change public method signature according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/)

Reply:
@jlnarvaez more specific: "inside of `getStringCustomAttributes()` will try to do a foreach to a null value" - just fix corresponding loop signature.

Reply:
@orlangur Ok, we check it in `getStringCustomAttributes()` method in new commit.